src/PowerShellEditorServices/Services/PowerShellContext/Handlers/GetVersionHandler.cs

```diff
@@ -52,5 +75,68 @@ private enum PowerShellProcessArchitecture
+            {
+                // The user has a good enough version of PackageManagement
+                if(module.Version >= s_desiredPackageManagementVersion)
+                {
+                    break;
+                }
+
+                _logger.LogDebug("Old version of PackageManagement detected. Attempting to update.");
+
+                var takeActionText = "Yes";
+                MessageActionItem messageAction = await _languageServer.Window.ShowMessage(new ShowMessageRequestParams
+                {
+                    Message = "You have an older version of PackageManagement known to cause issues with the PowerShell extension. Would you like to update PackageManagement (You will need to restart the PowerShell extension after)?",
+                    Type = MessageType.Warning,
+                    Actions = new []
+                    {
+                        new MessageActionItem
+                        {
+                            Title = takeActionText
+                        },
+                        new MessageActionItem
+                        {
+                            Title = "Not now"
+                        }
+                    }
+                });
+
+                // If the user chose "Not now" ignore it for the rest of the session.
+                if (messageAction?.Title == takeActionText)
+                {
+                    StringBuilder errors = new StringBuilder();
+                    await _powerShellContextService.ExecuteScriptStringAsync(
+                        "powershell.exe -NoLogo -NoProfile -Command 'Install-Module -Name PackageManagement -Force -MinimumVersion 1.4.6 -Scope CurrentUser -AllowClobber'",
+                        errors,
+                        writeInputToHost: true,
+                        writeOutputToHost: true,
+                        addToHistory: true).ConfigureAwait(false);
+
+                    // There were errors installing PackageManagement.
+                    if (errors.Length == 0)
+                    {
+                        _languageServer.Window.ShowMessage(new ShowMessageParams
+                        {
+                            Type = MessageType.Info,
+                            Message = "PackageManagement updated, If you already had PackageManagement loaded in your session, please restart the PowerShell extension."
+                        });
+                    }
+                    else
+                    {
+                        _languageServer.Window.ShowMessage(new ShowMessageParams
+                        {
+                            Type = MessageType.Error,
+                            Message = "PackageManagement update failed. Please run the following command in a Windows PowerShell prompt and restart the PowerShell extension: `Install-Module PackageManagement -Force -AllowClobber -MinimumVersion 1.4.6`"
```

Comment:
I think `Windows PowerShell prompt` here might be confusing. Perhaps `new powershell.exe console`? Or `a new Windows PowerShell session`?

Reply:
went with the latter.